posts/2022-01-14-construct-vendia-univseral-application-using-domain-driven-design.md

```diff
@@ -0,0 +1,177 @@
+* **Bounded Context**, has a one-one relationship with sub-domains. The sub-domain is the problem space(business) and the bounded context is the solution space(software). Bounded context is autonomous and has its own independent data models and architecture. It is where business meets the software systems. In the microservice world, a bounded context defines the maximum granularity of a service boundary. 
+* **Entities** are granular data structures such as order, customer, and product that have an identity and lifecycle outside of the application. Entities include fine-grained data structures such as personal details and contact details. 
+* **Scalars** are primitive data types such as phone number, email, etc. 
+* **Aggregates** are collections of Entities and Scalars. For example, the customer profile used by marketing is a combination of multiple entities such as customer demographics, order history, service requests, social activity, etc. Aggregates also have a top unique identifier called root entity reference. Often you will see aggregates referred to as a root entity.  For example, a customer profile will have a customerId as a root entity reference. 
+
+There are various ways in which you can come up with boundaries for your domains and sub-domains in your organization. You can use linguistics analysis, SME knowledge, business process, and business activity analysis. For better intuition on domains and sub-domains refer to generic domain modeling for retail e-commerce businesses. Figure 1 below describes a generic domain model of an e-Commerce business with domain boundaries and sub-domains inside it.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029034-fce0bc26-ac2b-4a0b-8034-839832160749.png" />
+</p>
+<p align="center">Figure 1 - eCommerce example domain model</p>
+
+
+## Business meets Software
+
+In the real world where everything is software with an existing install base, it is very hard to define a DDD bounded context. The way the business is organized and the way technology is implemented may not align well. In such cases, it becomes a three-legged race. It is more art than science. You can come up with several DDD models. But, the model that will prove useful is the one that will remove bottlenecks to improve software time to market and to service the core use case of the domain.
+
+Bounded Context is where the business meets software. Inside a bounded context, we go deep into the software architecture of an independent and autonomous business function and try to identify data aggregates, service boundaries for core use cases, and event definition for cross-domain synchronization. Figure 2 describes the Bounded Context Reference Architecture. 
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029167-ac2d28b7-7861-4790-a972-5a7a0ea50156.png" />
+</p>
+<p align="center">Figure 2 - Bounded Context Reference Architecture</p>
```

Comment:
@brian-vendia  - Did we land on a style convention for captions?  I've been doing  - centered, **Figure X** - _The Description_.   Happy to adopt whatever we'd like the norm to be.